test_infra/main.tf

```diff
@@ -0,0 +1,80 @@
+terraform {
```

Comment:
**suggestion:** @p-jankowiak-mrx How timeconsuming could be adding test suite for provider?  @timwardle This project could be developed by community, and more features and more fixes/bugs could make regression testing time consuming and not feasible. Also IMHO introducing new people to make changes here would be lot easier with unit tests of some sort. If that is not a best decision here for various reasons, then at least we should have some `Contribute` docs.

Reply:
I added few tests I think cover most of the simple cases. If we have more time we can develop it further